components/App/Auth/LoginForm.tsx

```diff
@@ -0,0 +1,307 @@
+import React from 'react';
+import AlertLayer from '../Game/AlertLayer';
+import {
+  Card,
+  Typography,
+  Button,
+  TextField,
+  Theme,
+  FormControl,
+  withStyles,
+  CardHeader,
+  IconButton,
+} from '@material-ui/core';
+import EmailIcon from '@material-ui/icons/Email';
+import VpnKeyIcon from '@material-ui/icons/VpnKey';
+import CloseIcon from '@material-ui/icons/Close';
+import AccountCircleIcon from '@material-ui/icons/AccountCircle';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import CardContent from '@material-ui/core/CardContent';
+import { LoginService, AUTH_RESULT_CODE, REG_RESULT_CODE } from './LoginService';
+
+const styles = (theme: Theme) => ({
+  margin: {
+    margin: theme.spacing(1),
+  },
+  cardHeader: {
+    padding: '0px',
+    paddingTop: '16px',
+  },
+});
+
+interface Props {
+  closeLoginForm: () => void;
+  classes: any; // FIXME
+}
+
+interface State {
+  email: string;
+  username: string;
+  password: string;
+  submitButtonEnabled: boolean;
+  loginStatus: AUTH_RESULT_CODE;
+  regStatus: REG_RESULT_CODE;
+  isRegistering: boolean;
+}
+
+class LoginForm extends React.Component<Props, State> {
+  emailInput: HTMLDivElement;
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: '',
+      email: '',
+      password: '',
+      submitButtonEnabled: false,
+      isRegistering: false,
```

Comment:
This is probably a bad idea. The UI looks very different from when the user is logging in and when they are registering.

It is better to have two classes, one for each case. And then the components that are re-used across them should be their own React components.